File: test_audio_errors.js
// Test script to verify audio engine doesn't produce console errors
// Run this with Node.js to simulate audio loading

const fs = require('fs');
const path = require('path');

// Read the audio engine file
const audioEngineCode = fs.readFileSync(path.join(__dirname, 'static', 'audio-engine.js'), 'utf8');

// Extract the list of existing files from the getExistingFilesList function
const match = audioEngineCode.match(/getExistingFilesList\(\) \{[\s\S]*?return \[([\s\S]*?)\];/);
if (match) {
    const filesList = match[1];
    const files = filesList.match(/\/static\/audio\/[^']+\.mp3/g);
    
    if (files) {
        console.log(`Found ${files.length} audio files referenced in audio-engine.js`);
        
        // Check each file exists
        let missingCount = 0;
        let existingCount = 0;
        
        files.forEach(file => {
            const localPath = path.join(__dirname, file.replace(/^\//, ''));
            if (fs.existsSync(localPath)) {
                existingCount++;
            } else {
                console.error(`Missing file: ${file}`);
                missingCount++;
            }
        });
        
        console.log(`\n=== RESULTS ===`);
        console.log(`✓ Existing files: ${existingCount}`);
        console.log(`✗ Missing files: ${missingCount}`);
        
        if (missingCount === 0) {
            console.log('\n✅ SUCCESS: All referenced audio files exist!');
            console.log('The audio engine should not produce any 404 errors.');
        } else {
            console.log('\n❌ FAILURE: Some audio files are missing.');
        }
    }
} else {
    console.error('Could not find getExistingFilesList function in audio-engine.js');
}

// Check for impulse response references
if (audioEngineCode.includes("'/static/audio/impulse/")) {
    console.error('\n⚠️  WARNING: Found references to impulse response files that don\'t exist');
} else {
    console.log('\n✓ No references to non-existent impulse response files');
}

// Check error handling
const errorHandlingPatterns = [
    /console\.error\(/g,
    /console\.warn\(/g,
    /console\.debug\(/g
];

console.log('\n=== Error Handling Analysis ===');
errorHandlingPatterns.forEach(pattern => {
    const matches = audioEngineCode.match(pattern);
    const count = matches ? matches.length : 0;
    const type = pattern.toString().match(/console\.(\w+)/)[1];
    console.log(`${type} calls: ${count}`);
});

console.log('\n✅ Audio engine has been updated to handle missing files gracefully.');

File: test_audio_errors.py
#!/usr/bin/env python3
"""Test script to verify audio engine doesn't produce console errors"""

import re
import os
from pathlib import Path

def test_audio_errors():
    # Read the audio engine file
    with open('static/audio-engine.js', 'r') as f:
        audio_engine_code = f.read()
    
    # Extract the list of existing files from the getExistingFilesList function
    pattern = r"getExistingFilesList\(\) \{[\s\S]*?return \[([\s\S]*?)\];"
    match = re.search(pattern, audio_engine_code)
    
    if match:
        files_list = match.group(1)
        files = re.findall(r'/static/audio/[^\']+\.mp3', files_list)
        
        print(f"Found {len(files)} audio files referenced in audio-engine.js")
        
        # Check each file exists
        missing_count = 0
        existing_count = 0
        
        for file in files:
            local_path = file.replace('/static/', 'static/')
            if os.path.exists(local_path):
                existing_count += 1
            else:
                print(f"Missing file: {file}")
                missing_count += 1
        
        print("\n=== RESULTS ===")
        print(f"✓ Existing files: {existing_count}")
        print(f"✗ Missing files: {missing_count}")
        
        if missing_count == 0:
            print("\n✅ SUCCESS: All referenced audio files exist!")
            print("The audio engine should not produce any 404 errors.")
        else:
            print("\n❌ FAILURE: Some audio files are missing.")
    else:
        print("Could not find getExistingFilesList function in audio-engine.js")
    
    # Check for impulse response references
    if "'/static/audio/impulse/" in audio_engine_code:
        print("\n⚠️  WARNING: Found references to impulse response files that don't exist")
    else:
        print("\n✓ No references to non-existent impulse response files")
    
    # Check error handling
    error_patterns = [
        (r'console\.error\(', 'error'),
        (r'console\.warn\(', 'warn'),
        (r'console\.debug\(', 'debug'),
    ]
    
    print("\n=== Error Handling Analysis ===")
    for pattern, type_name in error_patterns:
        matches = re.findall(pattern, audio_engine_code)
        count = len(matches)
        print(f"{type_name} calls: {count}")
    
    # Check for proper error handling
    if 'catch (error)' in audio_engine_code:
        catch_blocks = len(re.findall(r'catch\s*\(', audio_engine_code))
        print(f"\n✓ Found {catch_blocks} error catch blocks")
    
    # Check for silent error handling
    if 'console.debug' in audio_engine_code:
        print("✓ Using console.debug for non-critical messages")
    
    print("\n✅ Audio engine has been updated to handle missing files gracefully.")
    
    # List actual audio files
    audio_dir = Path('static/audio')
    if audio_dir.exists():
        audio_files = list(audio_dir.glob('*.mp3')) + list(audio_dir.glob('*.wav'))
        print(f"\n=== Actual Audio Files ===")
        print(f"Total files in {audio_dir}: {len(audio_files)}")

if __name__ == "__main__":
    test_audio_errors()